File: Solutions/Python/Product_Pair.py
class Solution:
    def isProduct(self, arr, n, x):
        # code here
        dict1 = dict(zip(arr, [0]*n))
        for i in range(n):
            if arr[i]!=0:
                if x/arr[i] in dict1:
                    return 1
        return 0

File: Solutions/Java/Product_Pair.java
class Solution {
    boolean isProduct(int[] arr, int n, long x) {
        // code here
        HashSet<Long> set = new HashSet<>();
        for (int i=0; i<n; i++) {
            if (arr[i]!=0 && x%arr[i]==0) {
                if (set.contains(x/arr[i])) {
                    return true;
                }
                else {
                    set.add(Long.valueOf(arr[i]));
                }
            }
        }
        return false;
    }
}
